services/user/Supervisor/ui/src/callStack.ts

```diff
@@ -0,0 +1,61 @@
+import { toString } from "@psibase/common-lib";
+import { QualifiedFunctionCallArgs } from "@psibase/common-lib/messaging";
+
+import { OriginationData } from "./utils";
+
+export interface Call {
+    caller: OriginationData;
+    args: QualifiedFunctionCallArgs;
+    startTime?: number;
+}
+
+// Callstack implementation that manages frames for every inter-plugin call
+export class CallStack {
+    private storage: Array<Call> = [];
+
+    push(sender: OriginationData, args: QualifiedFunctionCallArgs): void {
```

Comment:
Is it worth distinguishing these methods between `private` and `public`?
```suggestion
    public push(sender: OriginationData, args: QualifiedFunctionCallArgs): void {
```

Reply:
I was just marking when things were `private` since they are public by default. Although I see a few places where I applied this strategy inconsistently, so I've gone back and made sure all my classes use the strategy consistently.